async-gateway/service.go

```diff
@@ -77,32 +77,24 @@ func (s *service) CreateWorkload(id string, payload io.Reader, contentType strin
 
 // GetWorkload retrieves the status and result, if available, of a given workload
 func (s *service) GetWorkload(id string) (GetWorkloadResponse, error) {
-	prefix := s.workloadStoragePrefix()
 	log := s.logger.With(zap.String("id", id))
 
-	// download workload status
-	statusPath := fmt.Sprintf("%s/%s/status", prefix, id)
-	log.Debug("downloading status file", zap.String("path", statusPath))
-	statusBuf, err := s.storage.Download(statusPath)
+	status, err := s.getStatus(id)
 	if err != nil {
 		return GetWorkloadResponse{}, err
 	}
 
-	status := Status(statusBuf[:])
-	switch status {
-	case StatusFailed, StatusInProgress, StatusInQueue:
+	if status != StatusCompleted {
 		return GetWorkloadResponse{
 			ID:     id,
 			Status: status,
 		}, nil
-	case StatusCompleted: // continues execution after switch/case, below
-	default:
-		return GetWorkloadResponse{}, fmt.Errorf("invalid workload status: %s", status)
 	}
 
 	// attempt to download user result
+	prefix := s.workloadStoragePrefix()
```

Comment:
if the prefix is required below, compute it once and pass it as argument to `getStatus`

Reply:
Since `getStatus` has `s` as the receiver, and `s` has all the required information to compute the prefix, then I don't think we should pass the prefix into `getStatus` - it should be computed from the information that it has been given. Plus, I think that would make it a bit more convoluted.

Let me know what you think.

Reply:
that's double the allocations, but shouldn't be a problem, I guess.

`getStatus` is a helper method, so I'm not too worried about what it receives as argument